pkg/storer/internal/reserve/reserve.go

```diff
@@ -84,6 +86,8 @@ func New(
 
 // Put stores a new chunk in the reserve and returns if the reserve size should increase.
 func (r *Reserve) Put(ctx context.Context, store internal.Storage, chunk swarm.Chunk) (bool, error) {
+	r.putterMu.Lock()
```

Comment:
func that returns a new binID has a lock already

Reply:
if syncing started working "better" after this lock, then there is another hidden issue

Reply:
So we need the lock for the whole process of checking the chunkBinIndex. Parallel puller routines could be trying to add the same chunk and as only the binID increment is logged, it will end up adding duplicate items as we dont wait for the Put to finish before we check in Has.